src/user/services/user.service.spec.ts

```diff
@@ -22,4 +45,139 @@ describe('UserService', () => {
   it('should be defined', () => {
     expect(service).toBeDefined();
   });
+
+  describe('createUser', () => {
+    beforeEach(() => {
+      jest
+        .spyOn(bcrypt, 'hash')
+        .mockImplementation(async () => 'password-hashed');
+
+      jest
+        .spyOn(mockedUserRepository, 'save')
+        .mockImplementation(async () => mockedSavedUser);
+    });
+
+    it('should encrypt password before saving', async () => {
+      await service.createUser(mockedUserInput);
+      expect(bcrypt.hash).toBeCalledWith(mockedUserInput.password, 10);
+    });
+
+    it('should save user with encrypted password', async () => {
+      await service.createUser(mockedUserInput);
+      expect(mockedUserRepository.save).toBeCalledWith({
+        ...mockedUserInput,
+        password: 'password-hashed',
+      });
+    });
+
+    it('should return serialized user using `plainToClass`', async () => {
+      jest.spyOn(ct, 'plainToClass').mockImplementation(() => ({
```

Comment:
@asad-mlbd Any reason why we are mocking this function instead of letting the service function use it and return the real DTO class instance? :thinking:

Reply:
Actually, there is a weird problem with the repository save function. to solve that added in one test, then continued. Removed those, will describe in detail in the sync meeting.